### PR TITLE
Download feature for single episode with aria2c

### PR DESCRIPTION
**Pull request by ypawar159**
Hi @DxCx ,
My first commit to open source community!
I have added the download feature for single episode based on aria2 XML RPC interface.
This is initial work.
Definitely need to have more flexible options like entire series download or parts of it later on.

### CODE REVIEW

File: default.py [hunk 1]
```diff
@@ -103,7 +113,7 @@ def POPDUBBED_PAGES(payload):
 
 @route('search')
 def SEARCH(payload):
-    query = control.keyboard(control.lang(30007))
+    query = control.keyboard(control.lang(30002))
```
Comment (DxCx) [inline default.py]: please revert this change, 30007 is the correct value here.
it is refering to "Search" string:
https://github.com/DxCx/plugin.video.9anime/blob/c8a2c974f1eee1527624eb567ce329c71ee693d1/resources/language/English/strings.xml#L13

File: default.py [hunk 2]
```diff
@@ -4,8 +4,18 @@
 from resources.lib.ui.router import route, router_process
 from resources.lib.NineAnimeBrowser import NineAnimeBrowser
 import urlparse
+import xmlrpclib
```
Comment (DxCx) [inline default.py]: please remove imports and use `control.getSetting` instead.

File: default.py [hunk 3]
```diff
@@ -155,6 +166,25 @@ def PLAY(payload):
 def PLAY_SOURCE(payload):
     return control.play_source(urlparse.unquote(payload))
 
+@route('download*')
+def DOWNLOAD(payload):
+    url,name = payload.split("&name=")
```
Comment (DxCx) [inline default.py]: please cleanup and move to new function inside `control.py` `aria2_download(params....)`

File: resources/lib/ui/control.py [hunk 4]
```diff
@@ -55,7 +55,10 @@ def xbmc_add_player_item(name, url, iconimage=''):
     liz.setProperty("fanart_image", __settings__.getAddonInfo('path') + "/fanart.jpg")
     liz.setProperty("Video", "true")
     liz.setProperty("IsPlayable", "true")
-    liz.addContextMenuItems([], replaceItems=False)
+    #cmd = 'XBMC.Notification({},{})'.format("9anime",url.replace("playlink&url=/",""))
```
Comment (DxCx) [inline resources/lib/ui/control.py]: please cleanup,
also, instead of replacing playlink with download, we should add`&download=1` to download url `&download=0` for original url

then in playback route parse `download` flag value and either playitem or call aria2_download

**Comment by DxCx** [comment]
awsome work @ypawar159 , thanks alot for the contribution.
i've added comments inside on how i would make it better. :)

File: default.py (hunk 1 above)
Comment (ypawar159) [reply default.py]: I had not done any changes in search function.
It may have got from copy/pasting from another files I guess.
Will change it.

File: resources/lib/ui/control.py (hunk 4 above)
Comment (ypawar159) [reply resources/lib/ui/control.py]: Good suggestion.

**Comment by ypawar159** [comment]
Thanks @DxCx ,
I will try to implement the suggestions as time permits and update the PR.

**Comment by DxCx** [comment]
do you want me to try and complete it?

**Comment by ypawar159** [comment]
I will try to do it myself.  It will help me to learn a lot.

**Comment by ypawar159** [comment]
Now that I have a solution to the directory listing issue #204 , I can now work on the new code.
Now I know why branching is important.
Closing this PR. Will create a new one soon.
